src/app/auth/login/metamask.service.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { inject, Injectable, Injector } from "@angular/core";
+import { BrowserProvider } from "ethers";
+import { firstValueFrom, Observable } from "rxjs";
+import { SiweMessage } from "siwe";
+import { MaybeNull } from "src/app/interface/app.types";
+import { AuthApi } from "src/app/api/auth.api";
+import { ToastrService } from "ngx-toastr";
+import { Web3WalletCredentials } from "src/app/api/auth.api.model";
+
+@Injectable({
+    providedIn: "root",
+})
+export class MetamaskService {
```

Comment:
We don't necessarily have to use only Metamask as a wallet, can be any supporting EIP-1193

Proposed naming: `Eip1193EthereumService`

Reply:
Fixed